Merge nicholas-fiori/deployments-rails-add-pages-and-functionality from Sprint 1

Nick-CS-1 opened the pull request:
This merge covers partially issue #5 and merges branch nicholas-fiori/deployments-rails-add-pages-and-functionality in github/MLite

1. Added ability to create deployments
2. Added ability to view a list of deployments
3. Added ability to stop deployments
4. Status will display pending as default
5. Deployment link will generate its own link

Nick-CS-1 committed:
Added the ability to create new, view, and stop deployments.

1. Added ability to create deployments
2. Added ability to view a list of deployments
3. Added ability to stop deployments
4. Status will display pending as default
5. Deployment link will generate its own link

Issue #5

congbrian reviewed (commented):
**Code Functionality:** Code functionality seems fine.
**Code Readability:** Could have more comments on longer code blocks.
**Code Design:** Seems fine to me.
**Code Efficiency:** Not sure where changes in efficiency could even be made, considering that most code is just boilerplate frontend.
**Test Coverage:** Not seeing any tests implemented.
**GitHub Flow Practices:** Ensure that GitHub Flow is followed properly. Specifically, check that:
      _Branch Names:_ Seems fine. Is boilerplate code, doesn't require anything crazy.
      _Commit Messages:_  Seems fine. Could be a little more detailed given the amount of changes and code being committed, in terms of how code functions, but should be adequate.
      _Pull Request Descriptions:_ Seems fine to me.

Nick-CS-1 committed:
Added the ability to test using rails test

1. Added ability to test using rails built-in testing

Issue #5

Nick-CS-1 committed:
remove redundant .idea files and updated the gitignore

Nick-CS-1 committed:
Merge main

Issue #5

Nick-CS-1 committed:
Added model_id to deployment table

1. Added model_id in migration
2. Added relation code for ActiveRecord in models/
3. Updated Schema
4. Added user auth force for both deployments pages

Issue #5

Mohammad4844 reviewed (approved):
Tests pass. Functionality is as expected. Everything works.
